cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -355,61 +356,65 @@ private boolean appVersionFilter(boolean allowed, Banner banner) {
         try {
             PackageInfo pInfo = this.getCurrentActivity().getPackageManager().getPackageInfo(this.getCurrentActivity().getPackageName(), 0);
             String versionName = pInfo.versionName;
-            return this.checkRelationFilter(allowed, banner.getBannerAppVersionFilterRelation(), versionName, banner.getAppVersionFilterValue());
+            return this.checkRelationFilter(allowed, banner.getBannerAppVersionFilterRelation(), versionName, banner.getAppVersionFilterValue(), banner.getFromVersion(), banner.getToVersion());
         } catch (Exception e) {
             e.printStackTrace();
             Logger.e(TAG, "Error checking app version filter", e);
         }
+
         return allowed;
     }
 
-    private boolean checkRelationFilter(boolean allowed, CheckFilterRelation relation, String versionName, String compareValue) {
+    private boolean checkRelationFilter(boolean allowed, CheckFilterRelation relation, String appVersion, String compareValue, String fromVersion, String toVersion) {
         VersionComparator vc = new VersionComparator();
-        int mainValue = vc.compare(versionName, compareValue);
+
+        VersionComparison result = vc.compare(appVersion, compareValue);
+        VersionComparison resultFrom = vc.compare(appVersion, fromVersion);
+        VersionComparison resultTo = vc.compare(appVersion, toVersion);
 
         if (relation == null) {
             return allowed;
         }
 
         try {
             if (allowed && relation.equals(CheckFilterRelation.Equals)) {
-                if (mainValue != 0) {
+                if (result != VersionComparison.EQUALS) {
                     allowed = false;
                 }
             }
 
-            if (allowed && relation.equals(CheckFilterRelation.NotEqual)) {
-                if (mainValue == 0) {
+            if (allowed && relation.equals(CheckFilterRelation.Between)) {
+                if (resultFrom != VersionComparison.EQUALS && resultFrom != VersionComparison.GREATER_THAN && resultTo != VersionComparison.EQUALS && resultTo != VersionComparison.GREATER_THAN) {
```

Comment:
multiple lines please
Also if its equal, is it inbetween? not sure how we define this @gbhrdt

Reply:
here we are checking all conditions if the current sdkVersion is in between the required version or not.